### PR TITLE
Use new where syntax in misc files

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Comment by tknopp** [comment]
Out of curious: Is this now the recommended form for parameterized functions?

**Comment by yuyichao** [comment]
The old version might be deprecated/replaced in 1.0 so that it can be used to define outer constructors.

**Comment by tknopp** [comment]
This is a pretty fundamental change. The old syntax was pretty readable IMHO. I thought that `where` is only required for complex relations that could not be expressed before.

**Comment by musm** [comment]
> This is a pretty fundamental change. The old syntax was pretty readable IMHO. I thought that where is only required for complex relations that could not be expressed before.

The new syntax helps clarify parameterized function and parameterized constructors and IMO is a lot more clear and less confusing now.

**Comment by tknopp** [comment]
Could you give a link where this has been decided? In my point of view the former syntax was clear and readable, whereas the trailing notation requires looking back and forth (i.e. is less readable)

**Comment by musm** [comment]
> Could you give a link where this has been decided? In my point of view the former syntax was clear and readable, whereas the trailing notation requires looking back and forth (i.e. is less readable)

Very heavily discussed here   https://github.com/JuliaLang/julia/pull/18457

The new rule is a lot more simple:  if there is a `where` then `{T}` is now a literal; e.g. things such as the following are a  lot less confusing in 0.6 with the `where` syntax
```julia
type X{T}; end
X{T}(::T) = X{T}()

type Y{T}; end
Y{T}(::T) where {T} = Y{T}()


```

**Comment by JeffBezanson** [comment]
What does the following definition do?

```
Complex{Float64}(x) = ...
```

It shouldn't be controversial that this looks like it defines a constructor for the type `Complex{Float64}`, with one argument of any type. But it doesn't do that --- it defines a constructor for the type `Complex`, with a static parameter called `Float64` that doesn't occur anywhere, so the method is not actually callable. This syntax was pretty clearly a mistake.

See discussion starting here: https://github.com/JuliaLang/julia/issues/10146#issuecomment-73837796
Also issue #11310 which has most of the discussion
A bit of  #8135

**Comment by tknopp** [comment]
I see partly understand this issue within constructors. But this is a quite local issue in contrast to parametrized functions that occur a lot more often than constructors. I had a look at the above patch and it seems that 100% of the changes did not involve a constructor.

**Comment by JeffBezanson** [comment]
That is largely because this syntax change has already been done for constructors.

**Comment by JeffBezanson** [comment]
You're right that non-constructors occur much more often than constructors, but the syntax should still be consistent. We don't want `A{B}` to define a constructor for `A{B}` if `A` is a type, and otherwise introduce a static parameter. That would be both confusing and unnecessarily difficult to implement.

**Comment by tknopp** [comment]
I see the consistency issue here. I am not sure if this outweighs the regression in readability. One could also see the `old` syntax as a terse form for the `new` syntax and constructors are simply too complex to allow for a terse form.

**Comment by JeffBezanson** [comment]
Not everybody agrees that there _is_ any regression in readability. For instance I found the examples in https://github.com/JuliaLang/julia/issues/11310#issuecomment-170894828 compelling in favor of the new syntax.

Arguably, the list of static parameters and their bounds is not the first thing you want to see. Both the old and new syntax group together (1) the arguments, (2) the static parameter bounds, so with either syntax you might have to look back and forth. There is also a new syntax that combines them:

```
f(x::Vector{<:Real}, y::Vector{<:Number}) = ...
```

In that case there is truly no looking back and forth.

Some also find that the `where` is a nice visual separation in what used to be a dense nest of punctuation, e.g. `(::Type{T}){T}(arg)` vs. `(::Type{T})(arg) where T`.

**Comment by tknopp** [comment]
ok, maybe I simply need to get used to that. Thanks for the explanation.

**Comment by BobPortmann** [comment]
So could the first line in this pull request
```
function indices{T,N}(A::AbstractArray{T,N}, d)
```
become
```
function indices(A::AbstractArray{<:Any,<:Any}, d)
```
instead of
```
function indices(A::AbstractArray{T,N}, d) where {T,N}
```

**Comment by musm** [comment]
No because we need `N` in the function https://github.com/JuliaLang/julia/pull/21428/files#diff-2264bb51acec4e7e2219a3cb1c733651R40

I suppose this would work
`function indices(A::AbstractArray{<:Any,N}, d) where N`
but I think it's more readable with both explicit in this particular instance  ?

**Comment by BobPortmann** [comment]
Yes, of course

But I do agree with @tknopp that the new syntax is overly verbose for ordinary functions.

**Comment by musm** [comment]
I think once you get used to the syntax you start appreciating it, give it a shot first before dismissing it. It took me awhile to get used to the current syntax and how it works, but now it feels natural, so I think  familiarity is strong impact on preference.
e.g. I really appreciate the new syntax for things such as
` *(x::T,y::T) where {T<:Integer} #....`
vs.
` *{T<:Integer}(x::T,y::T) #....`

**Comment by musm** [comment]
BTW I think I will this this PR as is for now and call it g2g on my end, so feel free to review.
For future updates I will post to new PRs to help make review easier.

**Comment by musm** [comment]
timed out on one of the travis builds, otherwise tests pass

**Comment by musm** [comment]
Should this be squashed?

**Comment by tkelman** [comment]
Whoa, this got very large. Could this be split up into smaller pieces?

**Comment by musm** [comment]
Theoretically yes, but I hope I don't have to.

**Comment by tkelman** [comment]
A series of smaller PR's is always much easier to review and merge than one large one.

**Comment by StefanKarpinski** [comment]
I think that if this passes tests and was done via an automatic transformation, we can just go ahead.

**Comment by musm** [comment]
Well it was done via auto transformation plus manual checks and modifications here and there. For example I chose a function and then did the transformation and then it spits out what it changed and then I double checked the long form functions since that sometimes has other conventions in the file.

**Comment by musm** [comment]
I agree it's easier to review smaller PRs but the code here isn't adding or modifying existing code functionality, so I hope it's fine in this case.

**Comment by StefanKarpinski** [comment]
Ok, if it's not done yet, how about stopping here. Those changes sound fairly minimally risky, so if tests pass, let's merge.

**Comment by tkelman** [comment]
Not without reviewing, please

File: base/sparse/sparsematrix.jl [hunk 1]
```diff
@@ -206,7 +206,7 @@ function sparse_compute_reshaped_colptr_and_rowval{Ti}(colptrS::Vector{Ti}, rowv
     end
 end
 
-function reinterpret{T,Tv,Ti,N}(::Type{T}, a::SparseMatrixCSC{Tv,Ti}, dims::NTuple{N,Int})
+function reinterpret(::Type{T}, a::SparseMatrixCSC{Tv,Ti}, dims::NTuple{N,Int}) where {T,Tv,Ti,N}
```
Comment (Sacha0) [inline base/sparse/sparsematrix.jl]: When long-form method definitions involve several type parameters, the compact style in this definition (`where {T,Tv,Ti,N}`) seems clearly preferable to `where T where Tv where Ti where N`. Consistency of style also being advantageous, after what number of type parameters should the compact style be preferred in long-form method definitions? Drawing the line after anything more than one type parameter seems a bit arbitrary. Thoughts?

File: base/sparse/sparsevector.jl [hunk 2]
```diff
@@ -243,7 +243,7 @@ function setindex!{Tv,Ti<:Integer}(x::SparseVector{Tv,Ti}, v::Tv, i::Ti)
     x
 end
 
-setindex!{Tv, Ti<:Integer}(x::SparseVector{Tv,Ti}, v, i::Integer) =
+setindex!(x::SparseVector{Tv,Ti}, v, i::Integer) where {Tv, Ti<:Integer} =
```
Comment (Sacha0) [inline base/sparse/sparsevector.jl]: Extra space in `where {Tv, Ti<:Integer}`?

File: base/sparse/sparsematrix.jl (hunk 1 above)
Comment (musm) [inline base/sparse/sparsematrix.jl]: For this PR, if the original file already had a long form with multiple `where` I stuck to that condition.

File: base/sparse/sparsevector.jl (hunk 2 above)
Comment (musm) [inline base/sparse/sparsevector.jl]: Might as well fix the spacing too,thanks

File: base/sparse/sparsevector.jl [hunk 3]
```diff
@@ -817,7 +817,7 @@ vec(x::AbstractSparseVector) = x
 copy(x::AbstractSparseVector) =
     SparseVector(length(x), copy(nonzeroinds(x)), copy(nonzeros(x)))
 
-function reinterpret{T,Tv}(::Type{T}, x::AbstractSparseVector{Tv})
+function reinterpret(::Type{T}, x::AbstractSparseVector{Tv}) where {T,Tv}
```
Comment (Sacha0) [inline base/sparse/sparsevector.jl]: Along the lines of my general comment above, elsewhere in this commit long-form method definitions with two type parameters consistently use the `where T where Tv` form. Perhaps either this signature should change to `where T where Tv`, or the other long-form method definitions with two type parameters should switch to the compact form?

File: base/linalg/matmul.jl [hunk 4]
```diff
@@ -328,18 +328,18 @@ function herk_wrapper!{T<:BlasReal}(C::Union{StridedMatrix{T}, StridedMatrix{Com
     return generic_matmatmul!(C,tA, tAt, A, A)
 end
 
-function gemm_wrapper{T<:BlasFloat}(tA::Char, tB::Char,
-                                    A::StridedVecOrMat{T},
-                                    B::StridedVecOrMat{T})
+function gemm_wrapper(tA::Char, tB::Char,
+                                A::StridedVecOrMat{T},
```
Comment (JeffBezanson) [inline base/linalg/matmul.jl]: Wrong indent here.

File: base/int.jl [hunk 5]
```diff
@@ -25,17 +25,17 @@ const BitUnsigned64T = Union{Type{UInt8}, Type{UInt16}, Type{UInt32}, Type{UInt6
...
 ## integer comparisons ##
 
-<{T<:BitSigned}(x::T, y::T)  = slt_int(x, y)
+(<)(x::T, y::T) where {T<:BitSigned}  = slt_int(x, y)
 
--(x::BitInteger)             = neg_int(x)
--{T<:BitInteger}(x::T, y::T) = sub_int(x, y)
-+{T<:BitInteger}(x::T, y::T) = add_int(x, y)
-*{T<:BitInteger}(x::T, y::T) = mul_int(x, y)
+(-)(x::BitInteger)             = neg_int(x)
```
Comment (JeffBezanson) [inline base/int.jl]: Should line up.

File: base/sparse/sparsematrix.jl [hunk 6]
```diff
@@ -163,7 +163,7 @@ end
 
 ## Reinterpret and Reshape
 
-function reinterpret(::Type{T}, a::SparseMatrixCSC{Tv}) where T where Tv
+function reinterpret(::Type{T}, a::SparseMatrixCSC{Tv}) where {T, Tv}
```
Comment (Sacha0) [inline base/sparse/sparsematrix.jl]: Extra space in `where {T, Tv}`?

File: base/sparse/sparsematrix.jl [hunk 7]
```diff
@@ -293,7 +293,7 @@ function copy!(A::SparseMatrixCSC, B::SparseMatrixCSC)
 end
 
 similar(S::SparseMatrixCSC, Tv::Type=eltype(S)) = SparseMatrixCSC(S.m, S.n, copy(S.colptr), copy(S.rowval), Array{Tv}(length(S.nzval)))
-function similar(S::SparseMatrixCSC, ::Type{Tv}, ::Type{Ti}) where Tv where Ti
+function similar(S::SparseMatrixCSC, ::Type{Tv}, ::Type{Ti}) where {Tv, Ti}
```
Comment (Sacha0) [inline base/sparse/sparsematrix.jl]: Likewise here, extra space in `where {Tv, Ti}`?

File: base/sparse/sparsematrix.jl (hunk 6 above)
Comment (musm) [inline base/sparse/sparsematrix.jl]: You might be looking at an earlier commit, since I fixed this later on.

File: base/linalg/triangular.jl [hunk 8]
```diff
@@ -324,7 +324,7 @@ function -(A::UnitUpperTriangular)
 end
 
 # copy and scale
-function copy!{T<:Union{UpperTriangular, UnitUpperTriangular}}(A::T, B::T)
+function copy!(A::T, B::T) where T<:Union{UpperTriangular, UnitUpperTriangular}
```
Comment (Sacha0) [inline base/linalg/triangular.jl]: Extra space in `where T<:Union{UpperTriangular, UnitUpperTriangular}`?

File: base/linalg/triangular.jl [hunk 9]
```diff
@@ -333,7 +333,7 @@ function copy!{T<:Union{UpperTriangular, UnitUpperTriangular}}(A::T, B::T)
     end
     return A
 end
-function copy!{T<:Union{LowerTriangular, UnitLowerTriangular}}(A::T, B::T)
+function copy!(A::T, B::T) where T<:Union{LowerTriangular, UnitLowerTriangular}
```
Comment (Sacha0) [inline base/linalg/triangular.jl]: Likewise here, extra space?

File: base/sparse/sparsevector.jl [hunk 10]
```diff
@@ -415,7 +415,7 @@ function copy!(A::SparseVector, B::SparseMatrixCSC)
     return A
 end
 
-copy!{TvB, TiB}(A::SparseMatrixCSC, B::SparseVector{TvB,TiB}) =
+copy!(A::SparseMatrixCSC, B::SparseVector{TvB,TiB}) where {TvB, TiB} =
```
Comment (Sacha0) [inline base/sparse/sparsevector.jl]: Likewise here, extra space?

File: base/linalg/bidiag.jl [hunk 11]
```diff
@@ -199,15 +199,15 @@ end
 convert{T}(::Type{Matrix}, A::Bidiagonal{T}) = convert(Matrix{T}, A)
 convert(::Type{Array}, A::Bidiagonal) = convert(Matrix, A)
 full(A::Bidiagonal) = convert(Array, A)
-promote_rule{T,S}(::Type{Matrix{T}}, ::Type{Bidiagonal{S}})=Matrix{promote_type(T,S)}
+promote_rule(::Type{Matrix{T}}, ::Type{Bidiagonal{S}}) where {T,S}=Matrix{promote_type(T,S)}
```
Comment (Sacha0) [inline base/linalg/bidiag.jl]: Perhaps fix the spacing around `=` while you're at it? :)

File: base/linalg/bidiag.jl [hunk 12]
```diff
@@ -199,15 +199,15 @@ end
...
 
 #Converting from Bidiagonal to Tridiagonal
 Tridiagonal(M::Bidiagonal{T}) where {T} = convert(Tridiagonal{T}, M)
 function convert{T}(::Type{Tridiagonal{T}}, A::Bidiagonal)
     z = zeros(T, size(A)[1]-1)
     A.isupper ? Tridiagonal(z, convert(Vector{T},A.dv), convert(Vector{T},A.ev)) : Tridiagonal(convert(Vector{T},A.ev), convert(Vector{T},A.dv), z)
 end
-promote_rule{T,S}(::Type{Tridiagonal{T}}, ::Type{Bidiagonal{S}})=Tridiagonal{promote_type(T,S)}
+promote_rule(::Type{Tridiagonal{T}}, ::Type{Bidiagonal{S}}) where {T,S}=Tridiagonal{promote_type(T,S)}
```
Comment (Sacha0) [inline base/linalg/bidiag.jl]: Likewise here, perhaps fix the spacing around the `=` while you're at it? :)

File: base/pair.jl [hunk 13]
```diff
@@ -35,5 +35,5 @@ function convert(::Type{Pair{A,B}}, x::Pair) where A where B
     Pair{A, B}(convert(A, x[1]), convert(B, x[2]))
 end
 
-promote_rule{A1, B1, A2, B2}(::Type{Pair{A1, B1}}, ::Type{Pair{A2, B2}}) =
+promote_rule(::Type{Pair{A1, B1}}, ::Type{Pair{A2, B2}}) where {A1, B1, A2, B2} =
```
Comment (Sacha0) [inline base/pair.jl]: Extra spaces?

File: base/dates/types.jl [hunk 14]
```diff
@@ -298,7 +298,7 @@ Base.eltype{T<:Period}(::Type{T}) = T
 Base.promote_rule(::Type{Date}, x::Type{DateTime}) = DateTime
 Base.isless{T<:TimeType}(x::T, y::T) = isless(value(x), value(y))
 Base.isless(x::TimeType, y::TimeType) = isless(Base.promote_noncircular(x, y)...)
-=={T<:TimeType}(x::T, y::T) = ==(value(x), value(y))
+==(x::T, y::T) where {T<:TimeType} = ==(value(x), value(y))
```
Comment (Sacha0) [inline base/dates/types.jl]: (Unrelated, but the `} = ==(` tripped up my wetware parser. Perhaps infix `==` would be better?)

File: base/docs/utils.jl [hunk 15]
```diff
@@ -72,7 +72,7 @@ print(io::IO, t::Text) = print(io, t.content)
 print(io::IO, t::Text{<:Function}) = t.content(io)
 show(io::IO, t::Text) = print(io, t)
 
-=={T<:Union{HTML, Text}}(t1::T, t2::T) = t1.content == t2.content
+==(t1::T, t2::T) where {T<:Union{HTML, Text}} = t1.content == t2.content
```
Comment (Sacha0) [inline base/docs/utils.jl]: Extra space?

File: base/test.jl [hunk 16]
```diff
@@ -1105,10 +1105,10 @@ end
 #
 # Raises an error if any columnwise vector norm exceeds err. Otherwise, returns
 # nothing.
-function test_approx_eq_modphase{S<:Real,T<:Real}(
+function test_approx_eq_modphase(
         a::StridedVecOrMat{S}, b::StridedVecOrMat{T},
         err = length(indices(a,1))^3*(eps(S)+eps(T))
-    )
+    ) where {S<:Real,T<:Real}
```
Comment (Sacha0) [inline base/test.jl]: Indent seems odd?

File: base/int.jl [hunk 17]
```diff
@@ -25,17 +25,17 @@ const BitUnsigned64T = Union{Type{UInt8}, Type{UInt16}, Type{UInt32}, Type{UInt6
 
 ## integer comparisons ##
 
-<{T<:BitSigned}(x::T, y::T)  = slt_int(x, y)
+(<)(x::T, y::T) where {T<:BitSigned}  = slt_int(x, y)
```
Comment (Sacha0) [inline base/int.jl]: Extra space or alignment here as well?

File: base/promotion.jl [hunk 18]
```diff
@@ -194,18 +194,18 @@ end
...
 end
 
 # promote numeric types T and S to typejoin(T,S) if T<:S or S<:T
 # for example this makes promote_type(Integer,Real) == Real without
 # promoting arbitrary pairs of numeric types to Number.
-promote_to_supertype{T<:Number          }(::Type{T}, ::Type{T}, ::Type{T}) = (@_pure_meta; T)
-promote_to_supertype{T<:Number,S<:Number}(::Type{T}, ::Type{S}, ::Type{T}) = (@_pure_meta; T)
-promote_to_supertype{T<:Number,S<:Number}(::Type{T}, ::Type{S}, ::Type{S}) = (@_pure_meta; S)
-promote_to_supertype{T<:Number,S<:Number}(::Type{T}, ::Type{S}, ::Type) =
+promote_to_supertype(::Type{T}, ::Type{T}, ::Type{T}) where {T<:Number          } = (@_pure_meta; T)
```
Comment (Sacha0) [inline base/promotion.jl]: Perhaps the injection of spaces for alignment would be better between the `}` and `=`?

File: base/linalg/lq.jl [hunk 19]
```diff
@@ -137,15 +137,15 @@ end
 
 ## Multiplication by Q
 ### QB
-A_mul_B!{T<:BlasFloat}(A::LQPackedQ{T}, B::StridedVecOrMat{T})   = LAPACK.ormlq!('L','N',A.factors,A.τ,B)
+A_mul_B!(A::LQPackedQ{T}, B::StridedVecOrMat{T}) where {T<:BlasFloat}   = LAPACK.ormlq!('L','N',A.factors,A.τ,B)
```
Comment (Sacha0) [inline base/linalg/lq.jl]: Extra spaces?

File: base/linalg/matmul.jl [hunk 20]
```diff
@@ -40,10 +40,10 @@ end
 
 # Dot products
 
-vecdot{T<:BlasReal}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dot(x, y)
-vecdot{T<:BlasComplex}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dotc(x, y)
+vecdot(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) where {T<:BlasReal} = BLAS.dot(x, y)
+vecdot(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) where {T<:BlasComplex} = BLAS.dotc(x, y)
 
-function dot{T<:BlasReal, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}})
+function dot(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}}) where {T<:BlasReal, TI<:Integer}
```
Comment (Sacha0) [inline base/linalg/matmul.jl]: Extra space?

File: base/linalg/matmul.jl [hunk 21]
```diff
@@ -56,7 +56,7 @@ function dot{T<:BlasReal, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Ran
     BLAS.dot(length(rx), pointer(x)+(first(rx)-1)*sizeof(T), step(rx), pointer(y)+(first(ry)-1)*sizeof(T), step(ry))
 end
 
-function dot{T<:BlasComplex, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}})
+function dot(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}}) where {T<:BlasComplex, TI<:Integer}
```
Comment (Sacha0) [inline base/linalg/matmul.jl]: Extra space?

File: base/essentials.jl [hunk 22]
```diff
@@ -133,18 +133,18 @@ function typename(a::Union)
 end
 typename(union::UnionAll) = typename(union.body)
 
-convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T}, x::Tuple{Any, Vararg{Any}}) =
+convert(::Type{T}, x::Tuple{Any, Vararg{Any}}) where {T<:Tuple{Any,Vararg{Any}}} =
```
Comment (Sacha0) [inline base/essentials.jl]: Inconsistent / extra space?

Comment (musm) [inline base/essentials.jl]: This space was already there but I might as well also fix this thanks

File: base/sparse/sparsematrix.jl (hunk 6 above)
Comment (Sacha0) [inline base/sparse/sparsematrix.jl]: Yes :). I went through the commits sequentially, so chances are I commented on a few other things similarly fixed in later commits. Please ignore as appropriate :).

File: base/essentials.jl (hunk 22 above)
Comment (Sacha0) [inline base/essentials.jl]: Cheers :)

File: base/dates/types.jl (hunk 14 above)
Comment (musm) [inline base/dates/types.jl]: agreed, but I think this is better for a separate PR

**Review by Sacha0 — APPROVED** [review]
Much thanks for slogging through this @musm! Generally looks great.

(Abstracting from this PR, developing a consistent style guideline for when to use e.g. `where {S,T}` rather than `where S where T` in long-form method definitions involving more than one type parameter seems worthwhile. After reading through this diff, I lean towards using the compact form consistently.)

**Comment by musm** [comment]
> (Abstracting from this PR, developing a consistent style guideline for when to use e.g. where {S,T} rather than where S where T in long-form method definitions involving more than one type parameter seems worthwhile. After reading through this diff, I lean towards using the compact form consistently.)

Agreed, although I don't have a strong pref on the convention, but that we should have some guideline

Also thank you for going through this. I pushed a commit to address reviews. Can you please take another look to give it a final g2g.

File: base/test.jl [hunk 23]
```diff
@@ -1129,11 +1127,10 @@ elsewhere.
...
 function detect_ambiguities(mods...;
-        imported::Bool = false,
-        ambiguous_bottom::Bool = false,
-        allow_bottom::Union{Bool,Void} = nothing
-    )
-    if allow_bottom != nothing
+                            imported::Bool = false,
+                            ambiguous_bottom::Bool = false,
+                            allow_bottom::Union{Bool,Void} = nothing)
+    if allow_bottom !== nothing
```
Comment (musm) [inline base/test.jl]: unrelated but shoould be  !== not !=

File: base/test.jl [hunk 24]
```diff
@@ -1129,11 +1127,10 @@ elsewhere.
...
 """
 function detect_ambiguities(mods...;
-        imported::Bool = false,
-        ambiguous_bottom::Bool = false,
-        allow_bottom::Union{Bool,Void} = nothing
-    )
-    if allow_bottom != nothing
+                            imported::Bool = false,
+                            ambiguous_bottom::Bool = false,
+                            allow_bottom::Union{Bool,Void} = nothing)
```
Comment (musm) [inline base/test.jl]: Is this how we break  spacing ?

File: base/int.jl (hunk 17 above)
Comment (musm) [inline base/int.jl]: no this is supposed to be   aligned with the code in the lines below

**Comment by Sacha0** [comment]
> I pushed a commit to address reviews. Can you please take another look to give it a final g2g.

Checked out the touchup commit (#32) --- lgtm! :)

> Should this be squashed?

Some squashing seems in order. Though how much squashing would be best I don't know. Perhaps someone wiser in this regard will weigh in?

**Comment by musm** [comment]
I think it might just make the most sense to squash all the commits, I can't see any benefit otherwise

**Comment by musm** [comment]
can we merge this now?

**Comment by tkelman** [comment]
The changes here in `deprecated.jl` made Lint.jl and LanguageServer.jl start failing their tests - https://github.com/JuliaCI/pkg.julialang.org/blob/3e484195ef2fcccae5f1447447c3590d142e570e/logs/Lint_0.6.log

That's a bug in those packages, but in the interest of not breaking things that were previously passing right before an RC, we should probably revert the part of this that changed deprecated.jl for now.

**Comment by musm** [comment]
cc @TotalVerb  for `Lint.jl`
is there a log for LanguageServer ?

**Comment by tkelman** [comment]
https://github.com/JuliaCI/pkg.julialang.org/blob/3e484195ef2fcccae5f1447447c3590d142e570e/logs/LanguageServer_0.6.log but it's the exact same failure, I think it tries to Lint itself

**Comment by TotalVerb** [comment]
I'll have a fix in Lint. It's probably that lint is trying to parse Base's deprecated.jl file, and doesn't understand `where`.

**Comment by musm** [comment]
tahnks @TotalVerb  !

**Pull request merged**